doc/languages-frameworks/javascript.section.md

```diff
@@ -330,21 +330,25 @@ mkYarnPackage rec {
   - The `echo 9` steps comes from this answer: <https://stackoverflow.com/a/49139496>
   - Exporting the headers in `npm_config_nodedir` comes from this issue: <https://github.com/nodejs/node-gyp/issues/1191#issuecomment-301243919>
 
-## Outside of nixpkgs {#javascript-outside-nixpkgs}
+## Outside of Nixpkgs {#javascript-outside-nixpkgs}
 
-There are some other options available that can't be used inside nixpkgs. Those other options are written in Nix. Importing them in nixpkgs will require moving the source code into nixpkgs. Using [Import From Derivation](https://nixos.wiki/wiki/Import_From_Derivation) is not allowed in Hydra at present. If you are packaging something outside nixpkgs, those can be considered
+There are some other tools available, which are written in the Nix language.
+These that can't be used inside Nixpkgs because they require [Import From Derivation](#ssec-import-from-derivation), which is not allowed in Nixpkgs.
+Using them in Nixpkgs requires checking the intermediate build results into version control.
```

Comment:
I don't think it's possible to use them at all unless they are copy-pasted into nixpkgs
```suggestion
```

Reply:
A more accurate/clear description instead of intermediate would be "files from the source," I think.

Reply:
By `they` I meant to say npmlock2nix and nix-npm-buildpackage

Reply:
Right, but that's not related to the section I highlighted, unless I'm misunderstanding.

I'm saying that we can keep it if we reword it to be more accurate/clear.

Reply:
@winterqt not sure what you mean there. How would you reword the sentence?

@figsoda why would one have to paste the tools themselves into Nixpkgs? They could be a build dependency like any other, and come from anywhere. The problem here is that they will produce files which will have to be read during evaluation, which constitutes IFD. To work around that, you'll have to paste *those files* into Nixpkgs and check them into version control.

Reply:
IIUC both libraries are written in nix and you can't use them without ifd in nixokgs

Reply:
Yes, that is exactly what the blurb is trying to say. If the message doesn't come across I'm happy to accept suggestions.

Reply:
I feel like we're all talking about different things here.

@figsoda is talking about how we can't use them from their VCS repos without IFD, which is true -- but nothing would stop us from copying them over to here, and developing them further here.

I'm talking about how both mentioned tools require the use of IFD anyways, because they depend on parsing files in the source repository to generate the required Nix expressions. I think my angle is the better one to specify in the docs (over the "intermediate build artifacts" wording), since it's not something we can work around without copying source files from all packages.

I hope that makes sense 😅

Reply:
Ah okay, sorry. I was unaware that there is no explicit intermediate step that would allow avoiding IFD. Removed the sentence as suggested. I think it's irrelevant here that using them without IFD would require a fork. Well, of course, anything can be done if one rewrites the tool.

I hope this is correct now?